### PR TITLE
feat: update WakuPeerStore to store pubSubTopics for a peer

### PR DESCRIPTION
**Pull request by chaitanyaprem**
# Description
Store pubSubTopics that a peer supports. This is done keeping static sharding in mind as per  #727 
For autosharding, this can be reused, whereas API updates would be requried to only accept contentTopics.

Will take up sharded peer-mgmt in a separate PR in order to avoid making this one too big :) 

# Changes

- [x] Update WakuPeerStoreImpl to provide additional methods to set, retrieve pubSubTopics for a peer
- [x] AddPeer shall take an optional list of pubSubTopics. If nothing is passed, ENR is used to derive topics for relaySharding. If ENR is not present/empty for the peer then defaultRelayTopic is set.
- [x] Static Peers are provided with all pubSubTopics passed in the config as well as derived from content-topics(in case of auto-sharding). This means configuration should be changed to make it more fine-grain so that a static-Node can be specified which topics it supports.
- [x] Peers discovered with discv5 , PeerExchange, DNSDiscovery shall have ENR based on which pubSubTopics are derived.
- [x] Provide API's to fetch  peers based on pubSubTopic

# Tests

All unit tests have passed.
All builds have passed.

Note: Did not add new tests for these, as existing tests for discovery protocols seem to cover all flows.

### CODE REVIEW

File: waku/v2/peerstore/waku_peer_store.go [hunk 1]
```diff
@@ -139,3 +145,43 @@ func (ps *WakuPeerstoreImpl) Direction(p peer.ID) (network.Direction, error) {
...
+	return ps.peerStore.Put(p, peerPubSubTopics, existingTopics)
+}
+
+// SetPubSubTopics sets pubSubTopics for a peer, it also overrides existing ones that were set previously..
+func (ps *WakuPeerstoreImpl) SetPubSubTopics(p peer.ID, topics []string) error {
+	return ps.peerStore.Put(p, peerPubSubTopics, topics)
+}
+
+// PubSubTopics fetches list of pubSubTopics for a peer
+func (ps *WakuPeerstoreImpl) PubSubTopics(p peer.ID) ([]string, error) {
```
Comment (richard-ramos) [inline waku/v2/peerstore/waku_peer_store.go:165]: Would it make sense to check the shards stored in the ENRs of peers retrieved via discv5/peer exchange, and return these as well? (this question applies to `PeersByPubSubTopic` too)

**Review by richard-ramos — APPROVED** [review]
LGTM, however we'll probably need to add code in future PRs to maintain the peerstore, as peers might join/leave the pubsub topic. Looking at go-libp2p-pubsub code, I see there are some PeerJoin / PeerLeave events we can probably subscribe to, and this way determine whether a peer still is part of a pubsub topic or not

**Comment by chaitanyaprem** [comment]
> LGTM, however we'll probably need to add code in future PRs to maintain the peerstore, as peers might join/leave the pubsub topic. Looking at go-libp2p-pubsub code, I see there are some PeerJoin / PeerLeave events we can probably subscribe to, and this way determine whether a peer still is part of a pubsub topic or not

Right, that also needs to be handled. 
Let me add this as a task in the parent issue to keep track of this.

File: waku/v2/peerstore/waku_peer_store.go (hunk 1 above)
Comment (chaitanyaprem) [reply waku/v2/peerstore/waku_peer_store.go:165]: Those are already being translated to pubSubTopics and added when the peer is added to peerStore. 
https://github.com/waku-org/go-waku/blob/6f8933ef7971e83987f53f07b8ff89423ec964b2/waku/v2/peermanager/peer_manager.go#L219

https://github.com/waku-org/go-waku/blob/6f8933ef7971e83987f53f07b8ff89423ec964b2/waku/v2/discv5/discover.go#L492-L498

and 
https://github.com/waku-org/go-waku/blob/6f8933ef7971e83987f53f07b8ff89423ec964b2/waku/v2/protocol/peer_exchange/client.go#L111-L120

Unless i am missing something.

Comment (richard-ramos) [reply waku/v2/peerstore/waku_peer_store.go:165]: oh i see. I missed this!

**Pull request merged**